lib/register/index.js

```diff
@@ -0,0 +1,33 @@
+import check from '../check';
+import { code, message } from '../consts';
+
+/**
+ * Self register to window load
+ * Throw the error globally
+ */
+export default function register() {
+    function asynchronous() {
+
+        // Throw on next tick
+        setTimeout(() => {
+            const files = check();
+
+            if (files.length) {
+                const error = new Error([ message, ...files.map(({ src }) => src) ].join('\n'));
+                error.code = code;
+                error.files = files;
+
+                error.toJSON = () => ({
+                    message: error.message,
+                    code: error.code
+                });
+
+                throw error;
+            }
+        });
+    }
+
+    document.readyState === 'complete'
+        ? asynchronous()
+        : window.addEventListener('load', () => asynchronous);
```

Comment:
I think that you can do so:

window.addEventListener('load', asynchronous)

Without the need for the wrapping function.